js/src/components/app-modal/index.js

```diff
@@ -9,11 +9,36 @@ import classnames from 'classnames';
  */
 import './index.scss';
 
-const AppModal = ( props ) => {
-	const { buttons = [], className, children, ...rest } = props;
+const overflowYStyleName = {
+	auto: 'app-modal__styled--overflow-y-auto',
```

Comment:
❓  Why is the auto class necessary?

Reply:
The `overflow` of `.components-modal__frame` will be changed from `auto` to `hidden` in `@wordpress/components` **14.2.0**. I intended to make it have a stable overflow style across these versions. I added a code comment to clarify the intention in https://github.com/woocommerce/google-listings-and-ads/pull/1442/commits/1684e0e2469dbbc203f75214f137796cc1cb9f2c.

Reply:
❓ 

Yes, thats good, but I still find the extra class unnecessary, why not set ` overflow-Y: auto` to the modal without any need of that `app-modal__styled--overflow-y-auto`

Reply:
Since one of `overflow` axes is set to `visible` alone, it has a hidden rule in CSS spec that if the other axis is not `visible`, then the `visible` one is interpreted as `auto`.

I originally wanted to use CSS's own default value as much as possible and specify the Y-axis overflow only, but this also made the above rule implicit.

Considering there are added code comments and making it be a more explicit way, I changed the implementation in https://github.com/woocommerce/google-listings-and-ads/pull/1442/commits/28935f133511ea024a8844ac13a053ff47720102.